sqlmesh/core/engine_adapter/base.py

```diff
@@ -481,15 +529,70 @@ def _create_table_from_columns(
             if primary_key and self.SUPPORTS_INDEXES
             else []
         )
-        schema = exp.Schema(
+
+        schema = self._build_schema_exp(
+            table,
+            columns_to_types,
+            column_descriptions,
+            primary_key_expression,
+        )
+
+        self._create_table(
+            schema,
+            None,
+            exists=exists,
+            columns_to_types=columns_to_types,
+            table_description=table_description,
+            **kwargs,
+        )
+
+        # Register comments with commands if the engine doesn't support comments in the schema or CREATE
+        if (
+            table_description or column_descriptions
+        ) and self.COMMENT_CREATION == CommentCreation.COMMENT_COMMAND_ONLY:
+            self._create_comments(
+                table_name,
+                table_description,
+                column_descriptions,
+            )
+
+    def _build_schema_exp(
+        self,
+        table: exp.Table,
+        columns_to_types: t.Dict[str, exp.DataType],
+        column_descriptions: t.Optional[t.Dict[str, str]],
+        expressions: t.List[exp.PrimaryKey] = [],
+    ) -> exp.Schema:
+        """
+        Build a schema expression for a table, columns, column comments, and additional schema properties.
+        """
+        return exp.Schema(
             this=table,
             expressions=[
-                exp.ColumnDef(this=exp.to_identifier(column), kind=kind)
+                exp.ColumnDef(
+                    this=exp.to_identifier(column),
+                    kind=kind,
+                    constraints=self._build_col_comment_exp(column, column_descriptions)
+                    if column_descriptions
+                    and self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF
+                    else None,
+                )
                 for column, kind in columns_to_types.items()
             ]
-            + primary_key_expression,
+            + expressions,
         )
-        self._create_table(schema, None, exists=exists, columns_to_types=columns_to_types, **kwargs)
+
+    def _build_col_comment_exp(
+        self, col_name: str, column_descriptions: t.Dict[str, str]
+    ) -> t.List[exp.ColumnConstraint | None]:
```

Comment:
Doesn't look like we can return a list that contains `None` here ever

Reply:
Since we loop through all the columns on 580, this gets called for columns even if they don't have a comment in `column_descriptions` (so will return `[]` here).

Or am I misunderstanding?

Reply:
this method can never return a None, look at line 591

Reply:
Ah, I was misunderstanding - `t.List[exp.ColumnConstraint]` encompasses `[]` and doesn't need the `| None`